datastore/mongodb/error.go

```diff
@@ -45,3 +46,9 @@ func (e *er) Conflict() bool {
 func (e *er) Unavailable() bool {
 	return e.unavailable
 }
+
+// BadRequest satisfies the errBadRequest interface and allows other packages
+// to recall metadata about the error thrown
+func (e *er) BadRequest() bool {
```

Comment:
We don't want the mongo package to know anything specific to http directly, even though the existing stuff translates pretty well it's not a 1-to-1